added logic to fix map reset

DorianDeptuch opened the pull request:
Fixes #1919

  - [x] Up to date with `main` branch
  - [x] Branch name follows [guidelines](https://github.com/hackforla/311-data/blob/master/GETTING_STARTED.md#feature-branching)
  - [ ] All PR Status checks are successful
  - [ ] Peer reviewed and approved

Any questions? See the [getting started guide](https://github.com/hackforla/311-data/blob/master/GETTING_STARTED.md)

<details><summary>Screen recording  before fix</summary>

https://github.com/user-attachments/assets/57c4e372-a33d-4a02-af15-6c9477cbb40c

</details> 

<details><summary>Screen recording after fix</summary>

https://github.com/user-attachments/assets/306e85b9-9d3d-43ea-95a7-e45776e6a93e

</details> 

### Testing Instructions:

1.) The outside click reset should only trigger when an NC is selected
2.) Clicking outside without any NC selected should not cause the camera to zoom out

3.) Without Selecting an NC:

  - [ ] Load the map page

  - [ ] Click outside any NC boundaries

  - [ ] Observe that no camera reset or unintended behavior occurs

4.) After Selecting an NC:

  - [ ] Click on a specific NC to select it

  - [ ] Click outside the NC boundaries

  - [ ] Verify that the camera zooms out to the starting position

DrAcula27 reviewed (changes requested):
@DorianDeptuch, thanks for taking on this issue!
I think the change works as intended; however, can you add a brief description to the PR of what reviewers need to test for so we know if the code change worked?
- If I understood the issue, #1919, correctly, when no NC is selected & the user clicks outside any NC boundary (regardless of zoom level), nothing should happen; this is the behavior I am experiencing.
- Secondly, can you add, both in the original issue #1919 (for the product team) and in the PR (for us), a before and an after screenshot/video of the bug and the fix? 

Thanks!

DorianDeptuch commented:
@DrAcula27, thank you for pointing those out, I have added before and after screen recordings to both the PR and the original issue. I have also added testing instructions to clarify what the intended outcome is meant to be. Please let me know if there are any additional changes that I can make!

DrAcula27 reviewed (approved):
@DorianDeptuch Thanks for updating this! Your code passed the ABC's, including:
- PR is made from the correct branch
- PR links to the correct issue:
  - #1919
- The correct file `components/Map/Map.jsx` was edited
- No extra edits were made
- Appearance of the website is unchanged (Chrome and Firefox)
- Can confirm the expected behavior on my end as well (Chrome and Firefox)
Approved! 👍

ryanfchase reviewed (approved):
Good fix, thanks @DorianDeptuch
